Colour Ranked Status panel to the related status

tadatomix opened the pull request:
This PR was made by the same logic as #35145 and (hopefully) considers fixes, made by bdach. 

I decided to create this PR, after I saw this post #35148. The majority of the panels haven't got me into any issue, except graveyard one. I've tried to fix this issue and commented my reasonings.

I can't give a normal comparison images, since dev.ppy.sh don't have the maps for all of the needed cases. I'll provide the screenshots from a TestScene

Comparison:

|[`master`](https://github.com/ppy/osu/tree/master)|[`this PR`](https://github.com/tadatomix/osu/tree/song-select/coloured-group-ranked-status)|
|---|---|
|<img width="777" height="554" alt="изображение" src="https://github.com/user-attachments/assets/c453e765-17e7-4cff-9b8c-8a945f14305d" />| <img width="750" height="550" alt="new" src="https://github.com/user-attachments/assets/febbb71a-f8d7-4818-95d3-5cf83b05b006" />_Tried to stich the images as much as I could_|

tadatomix committed:
Make initial work to create a ranked status style

tadatomix committed:
Add a test for the new group

tadatomix committed:
Fix chevron being too bright in most cases

tadatomix committed:
Make Graveyard panel brighter

mcendu commented on osu.Game/Screens/SelectV2/PanelGroupRankedStatus.cs:
```diff
@@ -0,0 +1,217 @@
+            //Down bellow, I'll explain better the exact reasoning for this
+            switch (status)
+            {
+                //Graveyard pill was set to be fully black with some gray text.
+                //As long as it works for this case, this looks too bad on the coloured panel. (See -> https://github.com/ppy/osu/discussions/35148#discussioncomment-14609389)
+                //So my and OPs decision was to lighten it up, to make it look better
+                case BeatmapOnlineStatus.Graveyard:
+                    statusColour = new Color4(statusColour.R + 0.1f, statusColour.G + 0.1f, statusColour.B + 0.1f, 1); //That's quite the hacky way to achieve the needed result, but I haven't come up with a better decision. Lighten doesn't work at all.
+                    iconContainer.Colour = Color4.White;
+                    break;
```
Maybe `OsuColour.Gray6`? Or some other hue that should remind of black, while not being too dark.

tadatomix committed:
Change the pool of available Ranked Statuses

Co-authored-by: Bartłomiej Dach <dach.bartlomiej@gmail.com>

tadatomix committed:
Make Graveyard panel colour even brighter

tadatomix committed:
Remove unused OsuColor variable

bdach committed:
Remove strange comments

bdach committed:
Actually throw error instead of weirdly using white

bdach committed:
Change graveyard colour yet again

Editorial decision. The "brighter" colour was still too dark to be able
to even see any semblance of shade, or the triangles in the background.

bdach committed:
Adjust test

- Don't hardcode numerical enum bounds
- Exclude `Approved` as it doesn't show in real contexts

bdach committed:
Merge branch 'master' into song-select/coloured-group-ranked-status
